src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetPersonService.kt

```diff
@@ -105,8 +104,47 @@ class GetPersonService(
   }
 
   fun getPersonFromNomis(nomisNumber: String) = prisonerOffenderSearchGateway.getPrisonOffender(nomisNumber)
-}
 
-fun isNomsNumber(id: String?): Boolean {
-  return id?.matches(Regex("^[A-Z]\\d{4}[A-Z]{2}+$")) == true
+  fun getPrisoner(hmppsId: String): Response<Person?> {
+    val prisonerNomisNumber = getNomisNumber(hmppsId)
+
+    if (prisonerNomisNumber.errors.isNotEmpty()) {
+      return Response(
+        data = null,
+        errors = prisonerNomisNumber.errors,
+      )
+    }
+
+    val nomisNumber = prisonerNomisNumber.data?.nomisNumber
+    if (nomisNumber == null) {
```

Comment:
can be changed to     val nomisNumber = prisonerNomisNumber.data?.nomisNumber ?: return Response(
      data = null,
      errors = prisonerNomisNumber.errors,
    )
    
    also, is the check needed, as getNomisNumber() will return errors, is being handled at line 111